ios/FluentUI/Vnext/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift

```diff
@@ -0,0 +1,110 @@
+//
+
+import UIKit
+import SwiftUI
+
+/// UIKit wrapper that exposes the SwiftUI PersonaGrid implementation
+@objc open class MSFPersonaButtonCarousel: NSObject, FluentUIWindowProvider, MSFPersonaCarouselState {
+
+    // MARK: - Public API
+
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+
+    @objc public init(size: MSFPersonaButtonSize = .large,
+                      theme: FluentUIStyle? = nil) {
+        super.init()
+
+        personaButtonCarousel = PersonaButtonCarousel(size: size)
+
+        hostingController = UIHostingController(rootView: AnyView(personaButtonCarousel
+                                                                    .windowProvider(self)
+                                                                    .modifyIf(theme != nil, { personaButtonCarousel in
+                                                                        personaButtonCarousel.customTheme(theme!)
+                                                                    })))
+        hostingController.disableSafeAreaInsets()
+        view.backgroundColor = UIColor.clear
+    }
+
+    @objc public var count: Int {
+        return self.personaButtonCarousel.state.buttons.count
+    }
+
+    /// Adds a `PersonaButton` to the carousel, and returns an optional reference for additional property setting
+    ///
+    /// - Parameters:
+    ///   - primaryText: The primary text to appear in the `PersonaButton`
+    ///   - secondaryText: The secondary text to appear in the `PersonaButton`, below `primaryText`
+    ///   - image: The image to use as the persona's avatar
+    ///
+    /// - Returns: An optional reference to the added `PersonaButton`, which can be used to set additional properties or to update later
+    @discardableResult @objc public func add(primaryText: String?, secondaryText: String?, image: UIImage?) -> MSFPersonaButtonData {
+
+        let persona = MSFPersonaButtonStateImpl(size: self.state.buttonSize)
+
+        // Set passed-in properties
+        persona.primaryText = primaryText
+        persona.secondaryText = secondaryText
+        persona.image = image
+
+        self.personaButtonCarousel.state.buttons.append(persona)
+
+        return persona
+    }
+
+    /// Removes a `PersonaButton` from the carousel.
+    ///
+    /// - Parameters:
+    ///   - personaData: The reference to a `PersonaButton` to be removed.
+    @objc public func remove(_ personaData: MSFPersonaButtonData) {
+        self.personaButtonCarousel.state.buttons.removeAll { personaState in
+            personaState.isEqual(personaData)
+        }
+    }
+
+    /// Removes a `PersonaButton` from the carousel at the given index.
+    ///
+    /// - Parameters:
+    ///   - index: The index at which the `PersonaButton` to be removed can be currently found.
+    @objc public func remove(at index: Int) {
+        if index >= count {
+            fatalError("Attempting to remove item outside bounds of carousel")
```

Comment:
If I'm not mistaken, the AvatarGroup implements a no-op behavior in this case.
We should be consistent on this behavior across our collection states. 
Thoughts?

Reply:
I just checked and `AvatarGroup` also implements this as a `fatalError` so we're good.

Reply:
Regardless, I think firing a `fatalError` is safer than allowing the code to silently fail. If a client is trying to remove a `Persona` from a nonexistent index then it probably indicates an error in their own business logic somewhere.

Reply:
Sounds good.
I left a comment on the other PR so it's not updated to a no-op:
https://github.com/microsoft/fluentui-apple/pull/603/files#r658904309

@sophialee0416 FYI